Source: kde-packager@kde.org
Upstream: Yes.
Reason: Fix race condition resulting in ability to crash the lock dialog or even unlock it without providing any password.

SVN commit 1089241 by mitchell:
 
 As requested, now that distros have confirmed the fix for crash/hang of lock 
 dialog, backport to 4.4 branch. Pulled from r1089213.
 
 CCBUG: 217882
 
 M +10 -1 lockdlg.cc 
 M +1 -0 lockdlg.h 

--- branches/KDE/4.4/kdebase/workspace/krunner/lock/lockdlg.cc	(revision 1089240)
+++ branches/KDE/4.4/kdebase/workspace/krunner/lock/lockdlg.cc	(revision 1089241)
@@ -85,7 +85,8 @@
     : KDialog(parent, Qt::X11BypassWindowManagerHint),
       mPlugin( plugin ),
       mCapsLocked(-1),
-      mUnlockingFailed(false)
+      mUnlockingFailed(false),
+      sNot(0)
 {
     QWidget* w = mainWidget();
 
@@ -431,6 +432,8 @@
 //
 void PasswordDlg::gplugStart()
 {
+    if (sNot)
+        return;
     int sfd[2];
     char fdbuf[16];
 
@@ -461,6 +464,7 @@
     sFd = sfd[0];
     sNot = new QSocketNotifier(sFd, QSocketNotifier::Read, this);
     connect(sNot, SIGNAL(activated(int)), SLOT(handleVerify()));
+    connect(sNot, SIGNAL(destroyed()), SLOT(slotNotifierDestroyed()));
 }
 
 void PasswordDlg::gplugChanged()
@@ -647,6 +651,11 @@
         KDisplayManager().switchVT( itm->vt );
 }
 
+void PasswordDlg::slotNotifierDestroyed()
+{
+    sNot = 0;
+}
+
 void PasswordDlg::capsLocked()
 {
     unsigned int lmask;
--- branches/KDE/4.4/kdebase/workspace/krunner/lock/lockdlg.h	(revision 1089240)
+++ branches/KDE/4.4/kdebase/workspace/krunner/lock/lockdlg.h	(revision 1089241)
@@ -63,6 +63,7 @@
     void slotOK();
     void slotActivity();
     void handleVerify();
+    void slotNotifierDestroyed();
 
 private:
     void capsLocked();
